### PR TITLE
test: migrate second extensions-contrib batch to JUnit 5

### PR DESCRIPTION
**Pull request by FrankChen021**
Part of [#13948](https://github.com/apache/druid/issues/13948).

## Summary

This PR migrates the second independent `extensions-contrib` batch from JUnit 4 to JUnit 5:

- `moving-average-query`
- `aliyun-oss-extensions`
- `rabbit-stream-indexing-service`
- `opentsdb-emitter`

The module POMs remove JUnit 4 and Hamcrest test dependencies, and the test sources use Jupiter annotations, assertions, and parameterization. `MovingAverageQueryTest` now owns its small no-op scheduler fixture instead of loading the JUnit-4-based `QueryStackTests` class.

This branch is based directly on `master`; it does not stack on [#19908](https://github.com/apache/druid/pull/19908) and contains no shared SQL/processing fixture changes. It can be merged independently. The shared fixtures are intentionally owned only by #19908.

Related migration PRs: [#19908](https://github.com/apache/druid/pull/19908), [#19875](https://github.com/apache/druid/pull/19875), [#19876](https://github.com/apache/druid/pull/19876), [#19877](https://github.com/apache/druid/pull/19877), [#19878](https://github.com/apache/druid/pull/19878), [#19879](https://github.com/apache/druid/pull/19879), [#19880](https://github.com/apache/druid/pull/19880), [#19881](https://github.com/apache/druid/pull/19881), and [#19882](https://github.com/apache/druid/pull/19882).

## Validation

- `test-compile` passed for all four modules and their required reactor dependencies.
- Focused tests passed: moving-average (65), Aliyun OSS (68), Rabbit stream (39), and OpenTSDB (11).
- Checkstyle passed with 0 violations for all four modules.
- SpotBugs passed with 0 bugs/errors for all four modules.
- No JUnit 4 or Hamcrest source imports or direct POM dependencies remain in these modules.
- `git diff --check` passed.

### CODE REVIEW

File: extensions-contrib/aliyun-oss-extensions/src/test/java/org/apache/druid/storage/aliyun/OssDataSegmentMoverTest.java [hunk 1]
```diff
@@ -150,27 +154,29 @@
...
+              "baseKey/test/2013-01-01T00:00:00.000Z_2013-01-02T00:00:00.000Z/1/0/index.zip",
+              "bucket",
+              "DOES NOT EXIST"
+          ),
+          ImmutableList.of("dim1", "dim1"),
+          ImmutableList.of("metric1", "metric2"),
+          NoneShardSpec.instance(),
+          0,
+          1
+      ), ImmutableMap.of("bucket", "DOES NOT EXIST", "baseKey", "baseKey2"));
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-contrib/aliyun-oss-extensions/src/test/java/org/apache/druid/storage/aliyun/OssDataSegmentMoverTest.java:178]: Thanks for flagging this. This call predates the JUnit 5 migration; the PR only moved it into an assertThrows lambda. Migrating the deprecated DataSegment constructor to a builder is outside this PR’s scope, so I’m intentionally leaving it unchanged.

File: extensions-contrib/rabbit-stream-indexing-service/src/test/java/org/apache/druid/indexing/rabbitstream/RabbitStreamIndexTaskTuningConfigTest.java [hunk 2]
```diff
@@ -106,18 +101,18 @@
...
-    Assert.assertEquals(1000, (int) config.getRecordBufferSizeConfigured());
-    Assert.assertEquals(1000, config.getRecordBufferSizeOrDefault(1_000_000_000));
-    Assert.assertEquals(500, config.getRecordBufferOfferTimeout());
-    Assert.assertFalse(config.isResetOffsetAutomatically());
+    Assertions.assertNull(config.getBasePersistDirectory());
+    Assertions.assertEquals(new OnheapIncrementalIndex.Spec(), config.getAppendableIndexSpec());
+    Assertions.assertEquals(100, config.getMaxRowsInMemory());
+    Assertions.assertEquals(100, config.getMaxRowsPerSegment().intValue());
+    Assertions.assertEquals(new Period("PT1H"), config.getIntermediatePersistPeriod());
+    Assertions.assertEquals(100, config.getMaxPendingPersists());
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-contrib/rabbit-stream-indexing-service/src/test/java/org/apache/druid/indexing/rabbitstream/RabbitStreamIndexTaskTuningConfigTest.java:109]: Thanks for flagging this. This call predates the JUnit 5 migration; this PR only changes Assert to Assertions here. Deprecated API cleanup is outside this PR’s scope, so I’m intentionally leaving it unchanged.

File: extensions-contrib/rabbit-stream-indexing-service/src/test/java/org/apache/druid/indexing/rabbitstream/supervisor/RabbitStreamSupervisorTuningConfigTest.java [hunk 3]
```diff
@@ -134,22 +129,22 @@
...
-    Assert.assertEquals(17, (int) config.getMaxRecordsPerPollConfigured());
-    Assert.assertEquals(Duration.standardSeconds(15), config.getHttpTimeout());
-    Assert.assertEquals(Duration.standardSeconds(95), config.getShutdownTimeout());
-    Assert.assertEquals(Duration.standardSeconds(120), config.getRepartitionTransitionDuration());
+    Assertions.assertNull(config.getBasePersistDirectory());
+    Assertions.assertEquals(new OnheapIncrementalIndex.Spec(), config.getAppendableIndexSpec());
+    Assertions.assertEquals(100, config.getMaxRowsInMemory());
+    Assertions.assertEquals(100, config.getMaxRowsPerSegment().intValue());
+    Assertions.assertEquals(new Period("PT1H"), config.getIntermediatePersistPeriod());
+    Assertions.assertEquals(100, config.getMaxPendingPersists());
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-contrib/rabbit-stream-indexing-service/src/test/java/org/apache/druid/indexing/rabbitstream/supervisor/RabbitStreamSupervisorTuningConfigTest.java:137]: Thanks for flagging this. This call predates the JUnit 5 migration; this PR only changes Assert to Assertions here. Deprecated API cleanup is outside this PR’s scope, so I’m intentionally leaving it unchanged.

File: extensions-contrib/rabbit-stream-indexing-service/src/test/java/org/apache/druid/indexing/rabbitstream/supervisor/RabbitStreamSupervisorTuningConfigTest.java [hunk 4]
```diff
@@ -87,21 +82,21 @@
...
-    Assert.assertEquals(Duration.standardSeconds(10), config.getHttpTimeout());
-    Assert.assertEquals(Duration.standardSeconds(80), config.getShutdownTimeout());
-    Assert.assertEquals(Duration.standardSeconds(120), config.getRepartitionTransitionDuration());
-    Assert.assertEquals(100, config.getMaxRecordsPerPollOrDefault());
+    Assertions.assertNull(config.getBasePersistDirectory());
+    Assertions.assertEquals(new OnheapIncrementalIndex.Spec(), config.getAppendableIndexSpec());
+    Assertions.assertEquals(150000, config.getMaxRowsInMemory());
+    Assertions.assertEquals(5_000_000, config.getMaxRowsPerSegment().intValue());
+    Assertions.assertEquals(new Period("PT10M"), config.getIntermediatePersistPeriod());
+    Assertions.assertEquals(0, config.getMaxPendingPersists());
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-contrib/rabbit-stream-indexing-service/src/test/java/org/apache/druid/indexing/rabbitstream/supervisor/RabbitStreamSupervisorTuningConfigTest.java:90]: Thanks for flagging this. This call predates the JUnit 5 migration; this PR only changes Assert to Assertions here. Deprecated API cleanup is outside this PR’s scope, so I’m intentionally leaving it unchanged.

File: extensions-contrib/rabbit-stream-indexing-service/src/test/java/org/apache/druid/indexing/rabbitstream/RabbitStreamIndexTaskTuningConfigTest.java [hunk 5]
```diff
@@ -61,22 +56,22 @@
...
-    Assert.assertEquals(5000, config.getRecordBufferOfferTimeout());
-
-    Assert.assertFalse(config.isSkipSequenceNumberAvailabilityCheck());
-    Assert.assertFalse(config.isResetOffsetAutomatically());
+    Assertions.assertNull(config.getBasePersistDirectory());
+    Assertions.assertEquals(new OnheapIncrementalIndex.Spec(), config.getAppendableIndexSpec());
+    Assertions.assertEquals(150000, config.getMaxRowsInMemory());
+    Assertions.assertEquals(5_000_000, config.getMaxRowsPerSegment().intValue());
+    Assertions.assertEquals(new Period("PT10M"), config.getIntermediatePersistPeriod());
+    Assertions.assertEquals(0, config.getMaxPendingPersists());
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-contrib/rabbit-stream-indexing-service/src/test/java/org/apache/druid/indexing/rabbitstream/RabbitStreamIndexTaskTuningConfigTest.java:64]: Thanks for flagging this. This call predates the JUnit 5 migration; this PR only changes Assert to Assertions here. Deprecated API cleanup is outside this PR’s scope, so I’m intentionally leaving it unchanged.

**Review by FrankChen021 — COMMENTED** [review]
I have reviewed the code for correctness, edge cases, concurrency, and integration risks; no issues found.

Reviewed 53 of 53 changed files.

---

This is an automated review by Codex GPT-5.6-Luna(max)

File: extensions-contrib/aliyun-oss-extensions/src/test/java/org/apache/druid/storage/aliyun/OssDataSegmentPusherTest.java [hunk 6]
```diff
@@ -91,7 +90,7 @@ private void testPushInternal(boolean useUniquePath, String matcher) throws Exce
     OssDataSegmentPusher pusher = new OssDataSegmentPusher(client, config);
 
     // Create a mock segment on disk
-    File tmp = tempFolder.newFile("version.bin");
+    File tmp = new File(tempFolder, "version.bin");
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-contrib/aliyun-oss-extensions/src/test/java/org/apache/druid/storage/aliyun/OssDataSegmentPusherTest.java:94]: Fixed in commit `8a87f7204d`: made `tmp` `final`, satisfying AGENTS.md. The declaration is part of the modified JUnit 5 `@TempDir` migration line, so this is an in-scope style defect. Focused Aliyun Maven validation passed: compilation, checkstyle, and 13 tests (2 `OssDataSegmentPusherTest` + 11 `OssTaskLogsTest`).

File: extensions-contrib/aliyun-oss-extensions/src/test/java/org/apache/druid/storage/aliyun/OssTaskLogsTest.java [hunk 7]
```diff
@@ -424,7 +423,8 @@ private List<Grant> testPushInternal(boolean disableAcl, String ownerId, String
     OssTaskLogs taskLogs = new OssTaskLogs(ossClient, config, inputDataConfig, timeSupplier);
 
     String taskId = "index_test-datasource_2019-06-18T13:30:28.887Z";
-    File logFile = tempFolder.newFile("test_log_file");
+    File logFile = new File(tempFolder, "test_log_file");
+    Assertions.assertTrue(logFile.createNewFile());
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-contrib/aliyun-oss-extensions/src/test/java/org/apache/druid/storage/aliyun/OssTaskLogsTest.java]: Fixed in commit `8a87f7204d`: made `taskId` and `logFile` `final`, satisfying AGENTS.md. The modified `logFile` declaration is part of the JUnit 5 `@TempDir` migration; the earlier reply overlooked that line-level change. Focused Aliyun Maven validation passed: compilation, checkstyle, and 13 tests (2 `OssDataSegmentPusherTest` + 11 `OssTaskLogsTest`).

File: extensions-contrib/aliyun-oss-extensions/src/test/java/org/apache/druid/storage/aliyun/OssTimestampVersionedDataFinderTest.java [hunk 8]
```diff
@@ -102,7 +102,7 @@ public void testMissing()
 
     EasyMock.verify(oss);
 
-    Assert.assertEquals(null, latest);
+    Assertions.assertEquals(null, latest);
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-contrib/aliyun-oss-extensions/src/test/java/org/apache/druid/storage/aliyun/OssTimestampVersionedDataFinderTest.java:105]: Classified as pre-existing/context-only and outdated. Base commit `fe4377a4` already used `Assert.assertEquals(null, latest)` at this assertion; the migration only changed it to `Assertions.assertEquals(null, latest)`. This is a clarity improvement rather than a defect introduced by the JUnit 5 migration, so no code change is included.

File: extensions-contrib/aliyun-oss-extensions/src/test/java/org/apache/druid/data/input/aliyun/OssInputSourceTest.java [hunk 9]
```diff
@@ -538,20 +543,20 @@
...
-      Assert.assertEquals(NOW, nextRow.getTimestamp());
-      Assert.assertEquals("hello", nextRow.getDimension("dim1").get(0));
-      Assert.assertEquals("world", nextRow.getDimension("dim2").get(0));
+      Assertions.assertEquals(NOW, nextRow.getTimestamp());
+      Assertions.assertEquals("hello", nextRow.getDimension("dim1").get(0));
+      Assertions.assertEquals("world", nextRow.getDimension("dim2").get(0));
     }
 
-    Assert.assertEquals(2 * CONTENT.length, inputStats.getProcessedBytes());
+    Assertions.assertEquals(2 * CONTENT.length, inputStats.getProcessedBytes());
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-contrib/aliyun-oss-extensions/src/test/java/org/apache/druid/data/input/aliyun/OssInputSourceTest.java]: Fixed in commit `c4361e8ffb` by widening the operand before multiplication (`2L * CONTENT.length`), preventing the int overflow identified by CodeQL. The focused Aliyun OSS compilation, Checkstyle, and test validation passed. Resolving this fixed finding.

File: extensions-contrib/aliyun-oss-extensions/src/test/java/org/apache/druid/data/input/aliyun/OssInputSourceTest.java [hunk 10]
```diff
@@ -585,20 +590,20 @@
...
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-contrib/aliyun-oss-extensions/src/test/java/org/apache/druid/data/input/aliyun/OssInputSourceTest.java]: Fixed in commit `c4361e8ffb` by widening the operand before multiplication (`2L * CONTENT.length`), preventing the int overflow identified by CodeQL. The focused Aliyun OSS compilation, Checkstyle, and test validation passed. Resolving this fixed finding.

File: extensions-contrib/moving-average-query/src/test/java/org/apache/druid/query/movingaverage/MovingAverageQueryTest.java [hunk 11]
```diff
@@ -102,9 +101,17 @@
 /**
  * Base class for implementing MovingAverageQuery tests
  */
-@RunWith(Parameterized.class)
+@ParameterizedClass(name = "{0}")
+@MethodSource("data")
 public class MovingAverageQueryTest extends InitializedNullHandlingTest
 {
+  private static final QueryScheduler NOOP_SCHEDULER = new QueryScheduler(
```
Comment (FrankChen021) [inline extensions-contrib/moving-average-query/src/test/java/org/apache/druid/query/movingaverage/MovingAverageQueryTest.java:108]: It's OK to define a new scheduler here intead of depending on the one in other modules. This also eliminates the extra dependencies across test modules.

**Review by FrankChen021 — COMMENTED** [review]
manually reviewed all changes, LGTM

**Review by FrankChen021 — COMMENTED** [review]
I have reviewed the code for correctness, edge cases, concurrency, and integration risks; no issues found.

Reviewed 53 of 53 changed files.

The updated head introduces no static regression from the prior reviewed state.

Validation: read-only migration checks passed; tests were not rerun.



---

This is an automated review by Codex GPT-5.6-Luna(max)

**Review by GWphua — APPROVED** [review]
